playerbot/strategy/mage/GenericMageStrategy.cpp

```diff
@@ -44,8 +44,20 @@ class GenericMageStrategyActionNodeFactory : public NamedObjectFactory<ActionNod
             /*A*/ NextAction::array(0, new NextAction("shoot"), NULL),
             /*C*/ NULL);
     }
-    ACTION_NODE_A(frost_nova, "frost nova", "cone of cold");
-    ACTION_NODE_A(cone_of_cold, "cone of cold", "flee");
+    static ActionNode* frost_nova(PlayerbotAI* ai)
+    {
+        return new ActionNode("frost nova",
+            /*P*/ NULL,
+            /*A*/ NextAction::array(0, new NextAction("cone of cold"), NULL),
+            /*C*/ NULL);
+    }
+    static ActionNode* cone_of_cold(PlayerbotAI* ai)
+    {
+        return new ActionNode("cone of cold",
+            /*P*/ NULL,
+            /*A*/ NULL,
+            /*C*/ NULL);
+    }

```

Comment:
The flee action is managed by other triggers, ideally nothing should trigger a flee without a valid reason